### ISSUE TITLE
When installing blog-admin, occur a error "An error occurred while installing blog-admin (error: (error Package ‘org-8.0’ is unavailable))"

### ISSUE THREAD

**Issue by oabinga** [CodeFalling/blog-admin#11]

**Comment by xcodebuild** [comment]
How did you install blog-admin？From melpa？You have to upgrade your org-mode because blog-admin depends on org8.0+.

**Comment by oabinga** [comment]
Thank you.I have already inatlled org8.0+.I found the problem that I remove melpa from package-archives. After I add melpa, the problem is over.

**Comment by oabinga** [comment]
When I use blog-admin in spacemacs, there is a error:
Debugger entered--Lisp error: (invalid-read-syntax ") or . in a vector")
  read(get-file-char)
  edebug--read(read get-file-char)
  apply(edebug--read read get-file-char)
  #[128 "\300\301\302#\207" [apply edebug--read read nil] 5 nil](get-file-char)
  require(ox)
  eval-buffer(#<buffer  *load*-918356> nil "d:/home/.emacs.d/elpa/blog-admin-20160202.2203/blog-admin-backend-hexo.el" nil t)  ; Reading at buffer position 852
  load-with-code-conversion("d:/home/.emacs.d/elpa/blog-admin-20160202.2203/blog-admin-backend-hexo.el" "d:/home/.emacs.d/elpa/blog-admin-20160202.2203/blog-admin-backend-hexo.el" nil t)
  require(blog-admin-backend-hexo)
  eval-buffer(#<buffer  *load*-639371> nil "d:/home/.emacs.d/elpa/blog-admin-20160202.2203/blog-admin.el" nil t)  ; Reading at buffer position 1166
  load-with-code-conversion("d:/home/.emacs.d/elpa/blog-admin-20160202.2203/blog-admin.el" "d:/home/.emacs.d/elpa/blog-admin-20160202.2203/blog-admin.el" t t)
  require(blog-admin nil noerror)
  (not (require (quote blog-admin) nil (quote noerror)))
  (if (not (require (quote blog-admin) nil (quote noerror))) (ignore (message (format "Cannot load %s" (quote blog-admin)))))
  (prog1 (if (not (require (quote blog-admin) nil (quote noerror))) (ignore (message (format "Cannot load %s" (quote blog-admin))))) (let ((elapsed (float-time (time-subtract (current-time) now)))) (if (> elapsed 0.1) (message "%s...done (%.3fs)" "Loading package blog-admin" elapsed) (message "%s...done" "Loading package blog-admin"))))
  (let ((now (current-time))) (message "%s..." "Loading package blog-admin") (prog1 (if (not (require (quote blog-admin) nil (quote noerror))) (ignore (message (format "Cannot load %s" (quote blog-admin))))) (let ((elapsed (float-time (time-subtract (current-time) now)))) (if (> elapsed 0.1) (message "%s...done (%.3fs)" "Loading package blog-admin" elapsed) (message "%s...done" "Loading package blog-admin")))))
  (progn (if (condition-case err (run-hook-with-args-until-failure (quote use-package--blog-admin--pre-init-hook)) ((debug error) (ignore (display-warning (quote use-package) (format "%s %s: %s" "blog-admin" "pre-:init hook" (error-message-string err)) :error)))) (progn (condition-case err (progn (setq blog-admin-backend-path "d:/home/blog") (setq blog-admin-backend-type (quote hexo)) (setq blog-admin-backend-new-post-in-drafts t) (setq blog-admin-backend-new-post-with-same-name-dir t)) ((debug error) (ignore (display-warning (quote use-package) (format "%s %s: %s" "blog-admin" ":init" ...) :error)))) (condition-case err (run-hooks (quote use-package--blog-admin--post-init-hook)) ((debug error) (ignore (display-warning (quote use-package) (format "%s %s: %s" "blog-admin" "post-:init hook" ...) :error)))))) (let ((now (current-time))) (message "%s..." "Loading package blog-admin") (prog1 (if (not (require (quote blog-admin) nil (quote noerror))) (ignore (message (format "Cannot load %s" (quote blog-admin))))) (let ((elapsed (float-time (time-subtract ... now)))) (if (> elapsed 0.1) (message "%s...done (%.3fs)" "Loading package blog-admin" elapsed) (message "%s...done" "Loading package blog-admin"))))))
  my-org/init-blog-admin()
  funcall(my-org/init-blog-admin)

**Comment by xcodebuild** [comment]
@oabinga That should be another issue.How did you use `use` it in your spacemacs?And when did you see this error(at startup/`M-x blog-admin-start`?)

**Comment by oabinga** [comment]
The error occured at startup. First installed no error, after restart emacs, there is the error.
My conifg:
(defun my-org/init-blog-admin ()
  (use-package blog-admin
    :init
    (progn
    (setq blog-admin-backend-path "d:/home/blog")
    (setq blog-admin-backend-type 'hexo)
    (setq blog-admin-backend-new-post-in-drafts t) ;; create new post in drafts by default
    (setq blog-admin-backend-new-post-with-same-name-dir t) ;; create same-name directory with new post
    )))

**Comment by xcodebuild** [comment]
What's your emacs version? I am trying to reproduce this issue.The recommended version in Windows is `emacs-w64`

**Comment by xcodebuild** [comment]
And could you provide a list of your file in blog directory？

oabinga notifications@github.com于2016年3月16日周三 下午1:56写道：

**Comment by oabinga** [comment]
blog directory:

03-16 11:59 _config.yml  
01-08 11:24 db.json  
03-16 11:04 .deploy_git  
03-16 11:59 .git  
12-22 10:14 .gitignore  
03-16 11:57 node_modules 
03-16 11:57 package.json 
03-16 11:06 public  
03-16 11:06 scaffolds  
03-16 11:39 source  
03-16 11:06 src  
01-05 14:17 tail.md  
03-16 11:07 themes       

  d:/home/blog/source:  
11:16 404.html  
11:06 about  
11:06 categories 
11:53 _drafts  
11:06 images  
11:53 _posts  
11:06 tags

**Comment by oabinga** [comment]
My emacs is emacs-w64, and the version is 25.0.50.1.

**Comment by xcodebuild** [comment]
However I could not reproduce this issue. Especially `blog-admin` should do nothing at emacs startup.In your report it seems `require(ox)`  caused this.

**Comment by oabinga** [comment]
Thank you for your help. I will try to find out the problem.

**Comment by oabinga** [comment]
I think I have found the problem. When I installed blog-admin, emacs maybe downloaded a new org package that caused org-mode loading failed. I could not use org-mode for org file, before I deleted the package.Then I replace the new org package by older one, org mode worked and blog-admin was ok.I do not understand the reason. New package is org-20160314 and older package is org-20160118.

**Comment by xcodebuild** [comment]
`org-mode` do always break backward compatibility.I have no idea about this now. (I just fix another problem cause by meaningless break change.)
